### PR TITLE
Fixes #23320 - Add available_for "content_view_versions" to packages API

### PR DESCRIPTION
**Pull request by PaulSD**

### CODE REVIEW

**Comment by jlsherrill** [comment]
[test katello]

File: app/controllers/katello/api/v2/packages_controller.rb [hunk 1]
```diff
@@ -62,6 +62,38 @@ def custom_index_relation(collection)
...
 
+    api :GET, "/content_view_versions/:id/available_packages", N_("List packages that can be added to the Content View Version via an Incremental Update")
+    param :id, :number, :desc => N_("Content View Version identifier"), :required => true
+    param :repository_id, :number, :desc => N_("repository identifier")
+    param :organization_id, :number, :desc => N_("organization identifier to list applicable packages for")
+    param :host_id, :number, :desc => N_("host id to list applicable packages for")
+    param :packages_restrict_applicable, :bool, :desc => N_("show only packages with one or more applicable hosts")
+    param :packages_restrict_upgradable, :bool, :desc => N_("show only packages with one or more upgradable hosts")
+    param_group :search, ::Katello::Api::V2::ApiController
+    def available_packages
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/packages_controller.rb]: any objections to just putting all of this in the existing packages index action?  It already supports applicable, upgradable, host_id, and repository_id.  You'd just need to add some param 'available_for', as we've done in other places:  https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/host_collections_controller.rb#L34-L35
or
https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/products_controller.rb#L40

and then just put the logic in the 'indexed_relation' method.

Comment (jlsherrill) [reply app/controllers/katello/api/v2/packages_controller.rb]: oh i see, this is already a thing for the errata controller,  thoughts?

**Comment by PaulSD** [comment]
Re: https://github.com/Katello/katello/pull/7324#discussion_r183787294

I hadn't seen the available_for stuff...  I've updated this PR to use an available_for param on index instead of a separate available_packages method.

I wrote the available_errata method here: https://github.com/Katello/katello/pull/7050
For consistency, I'll open another PR to add support for available_for in the errata index too.  Can I remove the available_errata method, or do I need to leave it there for backward compatibility in case anyone has already started using it?

**Comment by jlsherrill** [comment]
@PaulSD yeah, it was released as part of 3.6, so i guess just mark it as deprecated similar to: https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/subscriptions_controller.rb#L67

Thanks!

**Pull request merged**